orderer/consensus/hlmirbft/chain.go

```diff
@@ -1467,8 +1467,36 @@ func (c *Chain) Apply(*msgs.QEntry) error {
 	return nil
 }
 
+//JIRA FLY2-66 proposed changes: Implemented the Snap Function
 func (c *Chain) Snap(networkConfig *msgs.NetworkState_Config, clientsState []*msgs.NetworkState_Client) ([]byte, []*msgs.Reconfiguration, error) {
-	return nil, nil, nil
+
+	var persistBytes []byte
+	c.Node.PurgeSnapFiles(c.opts.SnapDir)
```

Comment:
I don't think Snap should purge all snap files on every checkpoint as TransferTo may require them. It should just take a snapshot of the current network configuration. This function may be used by TransferTo so keep it for now but don't call it here.